ca/ca.go

```diff
@@ -52,14 +54,33 @@ type issuerMaps struct {
 	byNameID map[issuance.NameID]*issuance.Issuer
 }
 
+type CertProfileWithID struct {
+	// name is a human readable name used to refer to the certificate profile.
+	name string
+	// hash is SHA256 sum over every exported field of an issuance.ProfileConfig
+	// used to generate the embedded *issuance.Profile.
+	hash    [32]byte
+	profile *issuance.Profile
+}
+
+// certProfilesMap allows looking up the human-readable name of a certificate
+// profile to retrieve the actual profile.
+type certProfilesMaps struct {
+	profileByHash map[[32]byte]*CertProfileWithID
+	profileByName map[string]*CertProfileWithID
+}
+
 // certificateAuthorityImpl represents a CA that signs certificates.
 // It can sign OCSP responses as well, but only via delegation to an ocspImpl.
 type certificateAuthorityImpl struct {
 	capb.UnimplementedCertificateAuthorityServer
 	sa      sapb.StorageAuthorityCertificateClient
 	pa      core.PolicyAuthority
 	issuers issuerMaps
-	profile *issuance.Profile
+	// The name of the profile that will be selected if no explicit profile name
+	// is provided via gRPC.
+	defaultCertProfileName string
```

Comment:
Optional nit: if you like, you could store `defaultName` inside the `certProfileMaps`, so it doesn't have to be a separate field on the caImpl. This would also let you error out from `makeCertificateProfileMap` if the default name isn't represented among the options, rather than having to do that in issuePrecertificateInner.

Reply:
I have to pass the value an operator configures in `ca.json` into the CA though.

Reply:
Ohhhh ok I think I get it now.

Reply:
In the case of `issuePrecertificateInner` when the CA receives an empty name from a previously stored order in the database, should I have a map entry for `""`?

Reply:
I figured something out.